python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -512,6 +586,18 @@ def on_task_submitted(self, task_index: int, inputs: RefBundle):
         """Callback when the operator submits a task."""
         self.num_tasks_submitted += 1
         self.num_tasks_running += 1
+
+        if self._per_node_metrics_enabled:
+            node_ids = set()
+            for meta, node_metrics, node_id in iter_meta_and_node_metrics(
+                inputs, self._per_node_metrics
+            ):
+                if node_id not in node_ids:
+                    node_metrics.num_tasks_submitted += 1
+                    node_metrics.num_tasks_running += 1
+                # keep track of node ids to ensure we don't double count
+                node_ids.add(meta.exec_stats.node_id)
+
```

Comment:
See my prior comment -- you can't assume task will run where its input block _currently_ is

Instead you'd have to plug into `on_task_output_generated` to observe from where the task generate outputs though that would obviously delay us being able to capture the submission/running metrics.

cc @raulchen for ideas